File: vstemplates/templates-2005/CsSqlClrUdt/SqlClrUdt.cs
using System;
using System.Data.Sql;
using System.Data.SqlTypes;
using System.Runtime.InteropServices;
using Microsoft.SqlServer.Server;
using DM.Build.Yukon.Attributes;


[Serializable]
[StructLayout(LayoutKind.Sequential)]
[SqlUserDefinedType(Format.Native)]
public class $safeitemrootname$ : INullable {
  
  
  public override string ToString() {
    string s = "null";

    return s;
  }

  public bool IsNull {
    get {return true;}
  }

  public static $safeitemrootname$ Null {
    get {
      $safeitemrootname$ h = new $safeitemrootname$();

      return h;
    }
  }

  public static $safeitemrootname$ Parse(SqlString s) {
    if (s.IsNull || s.Value.ToLower() == "null")
      return Null;

    $safeitemrootname$ u = new $safeitemrootname$();

    return u;
  }

}

File: vstemplates/templates-2005/VBSqlClrUdt/SqlClrUdt.vb
Imports System
Imports System.Data.Sql
Imports System.Data.SqlTypes
Imports System.Runtime.InteropServices
Imports Microsoft.SqlServer.Server
Imports DM.Build.Yukon.Attributes


<Serializable()> _
<StructLayout(LayoutKind.Sequential)> _
<SqlUserDefinedType(Format.Native)> _
Public Class $safeitemrootname$
  Implements INullable


  Public Overrides Function ToString() As String
    Dim s = "null"
    'enter code

    Return s
  End Function

  Public ReadOnly Property IsNull() As Boolean Implements INullable.IsNull
    Get
      'enter code to decide what to return
      Return True
    End Get
  End Property

Public Shared ReadOnly Property Null As $safeitemname$
    Get
    Dim h as $safeitemrootname$ = New $safeitemrootname$()
      Return h
    End Get
  End Property
Public Shared Function Parse(ByVal s as SqlString) As $safeitemrootname$ 
    If (s.IsNull Or s.Value.ToLower() = "null") Then
      Return Null
    End If
  Dim u As $safeitemrootname$ = new $safeitemrootname$()
    'enter code


    Return u
  End Function

End Class

